pkg/subroutines/creator_test.go

```diff
@@ -0,0 +1,598 @@
+package subroutines_test
+
+import (
+	"context"
+	"testing"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openmfp/account-operator/api/v1alpha1"
+	"github.com/openmfp/account-operator/pkg/subroutines"
+	"github.com/openmfp/account-operator/pkg/subroutines/mocks"
+)
+
+type fgaError struct {
+	code codes.Code
+	msg  string
+}
+
+func (e fgaError) Error() string { return e.msg }
+func (e fgaError) GRPCStatus() *status.Status {
+	return status.New(e.code, e.msg)
+}
+
+func newFgaError(c openfgav1.ErrorCode, m string) *fgaError {
+	return &fgaError{
+		code: codes.Code(c),
+		msg:  m,
+	}
+}
+
+func TestCreatorSubroutine_GetName(t *testing.T) {
+	routine := subroutines.NewCreatorSubroutine(nil, nil, "")
+	assert.Equal(t, "CreatorSubroutine", routine.GetName())
+}
+
+func TestCreatorSubroutine_Finalizers(t *testing.T) {
+	routine := subroutines.NewCreatorSubroutine(nil, nil, "")
+	assert.Equal(t, []string{}, routine.Finalizers())
+}
+
+func TestCreatorSubroutine_Process(t *testing.T) {
+	namespace := "openmfp-system"
+	creator := "test-creator"
+
+	tests := []struct {
+		name        string
+		account     v1alpha1.Account
+		in          *openfgav1.WriteRequest
+		out         *openfgav1.WriteResponse
+		ctx         context.Context
+		setupMocks  func(context.Context, *openfgav1.WriteRequest, *openfgav1.WriteResponse, *mocks.OpenFGAServiceClient, *mocks.K8Service)
+		expectError *struct {
+			Retry  bool
+			Sentry bool
+		}
+	}{
+		{
+			name: "should_write_owner",
+			account: v1alpha1.Account{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-account",
+					Namespace: "openmfp-system",
```

Comment:
🟢 maybe use an account outside of openmfp-system, since this is the root namespace and the reconciler would probably get requests from accounts of different namespaces